pkg/registry/core/service/storage/rest_test.go

```diff
@@ -3892,7 +3892,7 @@ func TestDefaultingValidation(t *testing.T) {
 				},
 			},
 			expectedIPFamilyPolicy: &preferDualStack,
-			expectedIPFamilies:     []api.IPFamily{api.IPv4Protocol},
+			expectedIPFamilies:     []api.IPFamily{api.IPv4Protocol, api.IPv6Protocol},
```

Comment:
why if you "prefer" (not "require") and the cluster is single stack we have 2 ip families?

Reply:
Headless, selector-less semantics.

Reply:
Basically, this is a corner case we can't comprehend from Service alone.  It could already be holding dual-stack endpoints (I know some users were doing this).  So we acknowledge that and admit that this Service could have both.  "require" vs "prefer" was what we specced - it felt more explicit.